caffe2/operators/slice_op.cu

```diff
@@ -74,22 +74,23 @@ bool SliceImplGpu(
     if (i >= starts.numel()) {
       starts_idx[i] = 0;
       ends_idx[i] = data.size(i);
+      dst_sizes[i] = data.size(i);
       continue;
     }
     if (data.size(i) > 0) {
       auto start = starts_data[i];
       auto end = ends_data[i];
       if (start < 0) {
-        start = data.sizes()[i] + 1 + start;
+        start = data.size(i) + start;
```

Comment:
This actually changes the behavior of existing models and I don't think we can safely merge this.

We could define a complicated backward compatibility friendly rollout, but I don't think it's worth it given that Caffe2 frontend is not really actively developed.

I agree that the current logic of +1 everywhere doesn't make sense (and doesn't match numpy), but the proposed fix is even more confusing (as start and end will wrap around differently).

If you want - feel free to just update the documentation to describe this behavior.

Reply:
I agree that it is confusing in both versions. I just tried to make it compatible with current documentation which says that negative value for start index represents the number of elements before the end of that dimension. So -2 should represent two last elements instead of current one.
According to your suggestion to overcome this inconsistency from another side, I reverted this part and tried to update documentation.